packages/core/src/helpers/selector.ts

```diff
@@ -203,7 +204,7 @@ export function scopeNestedSelector(
                 : false;
             let nestedMixRoot = false;
             walkSelector(outputAst, (node, i, nodes) => {
-                if (node.type === 'nesting') {
+                if (isAnchor ? isAnchor(node) : node.type === 'nesting') {
```

Comment:
why the check of nesting still here?

Reply:
because for JS mixins we still use &. This will be the case until we deprecate v1 of mixins... first need to plan for v2...
I moved the default to the util args to make it easier to read